test/regression.tcl

```diff
@@ -230,14 +230,30 @@ proc run_test_lang { test lang } {
             set tmp_file [file join $result_dir $test.tmp]
             exec tr -d "\r" < $log_file > $tmp_file
             file rename -force $tmp_file $log_file
-            if [catch [concat exec diff $diff_options $ok_file $log_file \
-                         >> $diff_file]] {
-              puts " *FAIL*$error_msg"
-              append_failure $test
-              incr errors(fail)
+
```

Comment:
this proc is already quite long. Break out a new proc and give the proc a good name so that it is clear what is being tested. Testing for regressions in maximum resident set is bound to have very significant variations...